yarn.config.cjs

```diff
@@ -378,6 +379,31 @@ function expectWorkspaceField(workspace, fieldName, expectedValue = undefined) {
   }
 }
 
+/**
+ * Expect that the workspace has the given field, and that it is an array-like
+ * property containing the specified value. If the field is not present, is not
+ * an array, or does not contain the value, this will log an error, and cause
+ * the constraint to fail.
+ *
+ * @param {Workspace} workspace - The workspace to check.
+ * @param {string} fieldName - The field to check.
+ * @param {unknown} expectedValue - The value that should be contained in the array.
+ */
+function expectWorkspaceArrayField(workspace, fieldName, expectedValue) {
+  const fieldValue = get(workspace.manifest, fieldName);
+
+  if (!Array.isArray(fieldValue)) {
+    workspace.error(`Field "${fieldName}" is not an array.`);
+    return;
+  }
+
+  if (!fieldValue.includes(expectedValue)) {
+    workspace.error(
+      `Field "${fieldName}" does not contain the expected value "${expectedValue}".`,
+    );
+  }
```

Comment:
As indicated in JSDoc comment, we will validate if a given field "contains" the expected value (which includes array-like checks).

I'm erroring out, but I can refactor if we want it to be "fixable" where we set the field to be an array/append the array with the missing expected value.

Reply:
There are two key points to consider. If we don’t make this change fixable, it will modify the existing behavior, meaning developers will have to update the field manually. However, if we make it fixable, the change would closely resemble the functionality of expectWorkspaceField, with the main difference being the addition of validation to ensure the field is an array.

I think I understand the issue now: with the first approach, we would always set files to ['dist'], while in profile-sync-controller, we want it to be ["dist/", "auth/", "user-storage/"].

Given this, your added function makes sense. Perhaps we should make it fixable so that if fieldValue is undefined, it correctly sets the value to the expected one, preserving the original intent.

Reply:
Updated 👍🏾

Reply:
Thanks for the update. Could we also handle the case where `expectedValue` is undefined?

Reply:
@cryptodev-2s done.

Reply:
Thanks @Prithpal-Sooriya sorry I mean to handle case `expectedValue` and `fieldValue` is undefined the same way we do on `expectWorkspaceField `

Reply:
Cool, will be on it first thing tomorrow!

Reply:
@cryptodev-2s done!

We now only perform the automatic update when the `expectedValue` is provided. The `fieldValue` checks are still included (similar to the `expectWorkspaceField`).

Reply:
Great thanks @Prithpal-Sooriya sorry for the delay looks good now